New LGBM Version 4.0.0 changes

LGBM 4.0.0 was [released](https://github.com/microsoft/LightGBM/releases) 3 days ago. From [Friday runner](https://github.com/microsoft/hummingbird/actions/runs/5558411433/jobs/10153473207) `test_onnxml_lightgbm_converter.py`:

```
>       from lightgbm.basic import (
            _LIB, FEATURE_IMPORTANCE_TYPE_MAPPER, _safe_call,
            json_default_with_numpy)
E       ImportError: cannot import name 'FEATURE_IMPORTANCE_TYPE_MAPPER' from 'lightgbm.basic' (C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\lightgbm\basic.py)

C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\onnxmltools\convert\lightgbm\operator_converters\LightGbm.py:267: ImportError
```

They have added an underscore to many things. Ex: `FEATURE_IMPORTANCE_TYPE_MAPPER` becomes `_FEATURE_IMPORTANCE_TYPE_MAPPER`. See https://github.com/microsoft/LightGBM/pull/5654

Also in `test_onnxml_lightgbm_converter.py`:
```
FAILED tests/test_onnxml_lightgbm_converter.py::TestONNXLightGBMConverter::test_lightgbm_booster_classifier - AttributeError: 'Booster' object has no attribute 'attr'
```

Need to investigate what other breaking changes for us.

--- a/setup.py
+++ b/setup.py
@@ -31,13 +31,13 @@ install_requires = [
 ]
 onnx_requires = [
     "onnxruntime>=1.0.0",
-    "onnxmltools>=1.6.0,<=1.11.0",
+    "onnxmltools>=1.6.0",
     "skl2onnx>=1.7.0",
 ]
 extra_requires = [
     # The need each for these depends on which libraries you plan to convert from
     "xgboost>=0.90,<2.0.0",
-    "lightgbm>=2.2,<=3.3.5",
+    "lightgbm>=2.2",
     "holidays==0.24",
     "prophet==1.1",
 ]

